crates/next-api/src/server_actions.rs

```diff
@@ -167,7 +167,7 @@ async fn build_manifest(
             &key,
             ActionManifestWorkerEntry {
                 module_id: ActionManifestModuleId::String(loader_id.as_str()),
-                is_async: *chunk_item.is_self_async().await?,
+                is_async: *chunk_item.module().is_self_async().await?,
```

Comment:
Is self async is probably not enough. It should check if the item is async as derived from the graph

Reply:
Filed as PACK-3667